tests/components/knx/test_expose.py

```diff
@@ -123,3 +126,27 @@ async def test_expose_attribute_with_default(hass: HomeAssistant, knx: KNXTestKi
     # Change state to "off"; no attribute
     hass.states.async_set(entity_id, "off", {})
     await knx.assert_write("1/1/8", (0,))
+
+
+@patch("time.localtime")
+async def test_expose_with_date(localtime, hass: HomeAssistant, knx: KNXTestKit):
+    """Test an expose with a date."""
+    localtime.return_value = time.struct_time([2022, 1, 7, 9, 13, 14, 6, 0, 0])
+    await knx.setup_integration(
+        {
+            CONF_KNX_EXPOSE: {
+                CONF_TYPE: "datetime",
+                KNX_ADDRESS: "1/1/8",
+            }
+        },
+    )
+    assert not hass.states.async_all()
+
+    await knx.receive_read("1/1/8")
+    await knx.assert_write("1/1/8", (0x7A, 0x1, 0x7, 0xE9, 0xD, 0xE, 0x20, 0x80))
+    await knx.assert_response("1/1/8", (0x7A, 0x1, 0x7, 0xE9, 0xD, 0xE, 0x20, 0x80))
```

Comment:
Why do we have both, a write and a response? If the write is from init it should be assertable before the incoming read.

Reply:
Done!